lib/handlers/router-entities/route-caching/dynamo-route-caching-provider.ts

```diff
@@ -190,21 +205,25 @@ export class DynamoRouteCachingProvider extends IRouteCachingProvider {
     return undefined
   }
 
-  /**
-   * Implementation of the abstract method defined in `IRouteCachingProvider`
-   * Attempts to insert the `CachedRoutes` object into cache, if the CachingStrategy returns the CachingParameters
-   *
-   * @param cachedRoutes
-   * @param amount
-   * @protected
-   */
-  protected async _setCachedRoute(cachedRoutes: CachedRoutes, amount: CurrencyAmount<Currency>): Promise<boolean> {
+  public generateCachedRouteDbEntry(
+    cachedRoutes: CachedRoutes,
+    amount: CurrencyAmount<Currency>
+  ): CachedRouteDbEntry | null {
     const cachedRoutesStrategy = this.getCachedRoutesStrategyFromCachedRoutes(cachedRoutes)
     const cachingBucket = cachedRoutesStrategy?.getCachingBucket(amount)
+    const chainId = cachedRoutes.chainId
+    const blocksToLive = cachedRoutes.blocksToLive
+    let cachedRoutesTtl = this.ttlMinutes
+    const secondsToLivePerBlock = SECONDS_PER_BLOCK_BY_CHAIN_ID[chainId]
+
+    if (blocksToLive > 0 && typeof secondsToLivePerBlock === 'number') {
+      // use this.ttlMinutes as a buffer in addition to average calculated ttl for blocksToLive
+      cachedRoutesTtl = secondsToLivePerBlock * blocksToLive + this.ttlMinutes * 60
```

Comment:
I don't think we need the `ttlMinutes` here, I would probably add a buffer in the `SECONDS_PER_BLOCK_BY_CHAIN_ID` config instead.

Reply:
done ty, i doubled it for now